sources/packages/api/src/route-controllers/application/application.controller.ts

```diff
@@ -1,624 +1,84 @@
-      await this.disbursementScheduleService.getFirstCOEOfApplication(
-        applicationId,
-      );
-    return transformToApplicationDetailDto(application, firstCOE);
-  }
+    private readonly locationService: InstitutionLocationService,
+    private readonly programService: EducationProgramService,
+  ) {}
   /**
-   * Submit an existing student application changing the status
-   * to submitted and triggering the necessary processes.
-   * The system will ensure that an application will always be
-   * transitioning from draft to submitted status. The student
-   * application is not supposed to be created directly in the
-   * submitted status in any scenario.
-   * @param payload payload to create the draft application.
-   * @param applicationId application id to be changed to submitted.
-   * @param userToken token from the authenticated student.
+   * Add location, program and offering labels
+   * and reset the dropdown value for non
+   * designated location and not approved
+   * programs.
+   * @param data application data
+   * @returns [ApplicationData, ApplicationFormData]
    */
-  @CheckSinValidation()
-  @CheckRestrictions()
-  @AllowAuthorizedParty(AuthorizedParties.student)
-  @Patch(":applicationId/submit")
-  async submitApplication(
-    @Body() payload: SaveApplicationDto,
-    @Param("applicationId") applicationId: number,
-    @UserToken() userToken: IUserToken,
-  ): Promise<void> {
-    const programYear = await this.programYearService.getActiveProgramYear(
-      payload.programYearId,
-    );
-    if (!programYear) {
-      throw new UnprocessableEntityException(
-        "Program Year is not active. Not able to create an application invalid request",
-      );
-    }
-
-    const submissionResult = await this.formService.dryRunSubmission(
-      programYear.formName,
-      payload.data,
-    );
-    if (!submissionResult.valid) {
-      throw new BadRequestException(
-        "Not able to create an application due to an invalid request.",
-      );
-    }
-    // studyStartDate from payload is set as studyStartDate
-    let studyStartDate = payload.data.studystartDate;
-    let studyEndDate = payload.data.studyendDate;
-    if (payload.data.selectedOffering) {
-      const offering = await this.offeringService.getOfferingById(
-        payload.data.selectedOffering,
-      );
-      // if  studyStartDate is not in payload
-      // then selectedOffering will be there in payload,
-      // then study start date taken from offering
-      studyStartDate = offering.studyStartDate;
-      studyEndDate = offering.studyEndDate;
-    } else {
-      // when selectedOffering is not selected
-      const notValidDates = checkNotValidStudyPeriod(
-        payload.data.studystartDate,
-        payload.data.studyendDate,
-      );
-      if (notValidDates) {
-        throw new UnprocessableEntityException(
-          `${INVALID_STUDY_DATES} ${notValidDates}`,
+  async addLabelsAndResetDropdownForNotApproved(
```

Comment:
Please rename it to something not so specific to the UI.
IMO it could be renamed just as _generateApplicationFormData_.